pkg/sentry/socket/netlink/route/protocol.go

```diff
@@ -500,6 +501,80 @@ func (p *Protocol) delAddr(ctx context.Context, msg *netlink.Message, ms *netlin
 	return nil
 }
 
+func fetchRoute(msg *netlink.Message) (*inet.Route, *syserr.Error) {
+	var rtMsg linux.RouteMessage
+	attrs, ok := msg.GetData(&rtMsg)
+	if !ok {
+		return nil, syserr.ErrInvalidArgument
+	}
+	if rtMsg.Table != linux.RT_TABLE_MAIN {
+		// Only the main table since we don't have multiple
+		// routing tables.
+		return nil, syserr.ErrInvalidArgument
+	}
+	route := inet.Route{
+		Family:   rtMsg.Family,
+		DstLen:   rtMsg.DstLen,
+		SrcLen:   rtMsg.SrcLen,
+		TOS:      rtMsg.TOS,
+		Table:    rtMsg.Table,
+		Protocol: rtMsg.Protocol,
+		Scope:    rtMsg.Scope,
+		Type:     rtMsg.Type,
+		Flags:    rtMsg.Flags,
+	}
+
+	attHdr, buff, rest, ok := attrs.ParseFirst()
+	for ok {
+		switch attHdr.Type {
+		case linux.RTA_DST:
+			route.DstAddr = buff
+		case linux.RTA_SRC:
+			route.SrcAddr = buff
+		case linux.RTA_GATEWAY:
+			route.GatewayAddr = buff
+		case linux.RTA_OIF:
+			// This conversion is also present in the kernel...
+			var outputIF primitive.Int32
+			outputIF.UnmarshalUnsafe(buff)
+			route.OutputInterface = int32(outputIF)
+		default:
+			// Other attributes are not implemented yet.
+			return nil, syserr.ErrInvalidArgument
+		}
+		attHdr, buff, rest, ok = rest.ParseFirst()
+	}
+	return &route, nil
+}
+
+// newRoute handles RTM_NEWROUTE requests.
+func (p *Protocol) newRoute(ctx context.Context, msg *netlink.Message, ms *netlink.MessageSet) *syserr.Error {
+	stack := inet.StackFromContext(ctx)
+	if stack == nil {
+		// No network routes.
+		return nil
+	}
+	hdr := msg.Header()
+	flags := hdr.Flags&^(linux.NLM_F_REQUEST|linux.NLM_F_ACK)
+	if flags != linux.NLM_F_CREATE|linux.NLM_F_EXCL {
+		// ip route prepend/append/change/replace/test are not implemented.
```

Comment:
Please open a github issue and put a TODO here?

Reply:
These commands rarely ever used as far as I know, so I don't think an Issue is required :)
I'll add TODO there.

Reply:
You can just link it back to the base issue. so 
// TODO(gvisor.dev/issue/578)

Reply:
Sure. Just link it to that issue. It'll be fine.

Reply:
Fixed.